settings/settings-url.php

```diff
@@ -273,7 +273,8 @@ protected function update( $options ) {
 					);
 				}
 				else {
-					$newoptions['domains'][ $key ] = esc_url_raw( trim( $domain ) );
+					$domain = array( $key => esc_url_raw( trim( $domain ) ) );
+					$newoptions['domains'] = $domain;
```

Comment:
If you have several domains, each domain will be overwritten by the next one. So I suggest to leave this line alone and instead:
```php
	protected function update( $options ) {
		$newoptions = array(
			'domains' => array(),
		);
```

Reply:
:man_facepalming: 
Indeed, I didn't see that ...
I made something else inspired by your solution. Because what you proposed still raised an error.